lib/neo4j/active_node/has_n.rb

```diff
@@ -101,9 +114,12 @@ def cached?
       end
 
       def replace_with(*args)
-        @cached_result = nil
-
-        @query_proxy.public_send(:replace_with, *args)
+        nodes = @query_proxy.public_send(:replace_with, *args).to_a
```

Comment:
@cheerfulstoic  Really don't know. I left as it was 😆 We can change it, I think ;)